WordPress/gutenberg
# Add Port To `WP_TESTS_DOMAIN`

## What?
This PR adds the configured port to `WP_TESTS_DOMAIN` so that tests referencing `'http://' . WP_TESTS_DOMAIN . /path` will work with a port other than 80.

## Why?
I am currently working on running unit tests within the container, however, I have run into problems because the URLs generated by WordPress contain the port while `WP_TESTS_DOMAIN` does not. By adding the port to the constant we are able to make tests more resilient and better support environment customization.

I have looked at WordPress' tests and there doesn't seem to be anything wrong with just adding the port to the constant. This also removes the need for the port in Gutenberg's tests, which this PR removes.

## How?
This pull request refactors the way that ports are added to configuration options. This let me write some nice tests for different cases and handle inputs that aren't URLs. To keep the validation in place I have added the URL validation to the config validation function instead.

In the process of making these changes I also noticed that the `--url` being passed to WordPress' installation command has the port included twice. There's no reason to use anything other than `WP_SITEURL` since the constant will override it anyway. It will have the correct port and is the most accurate input to use here.

## Testing Instructions

1. Passing CI is a great indicator that the test-related functionality in this PR works.
2. Try running the PHPUnit suite locally with a `.wp-env.override.json` file that changes the port of the test container.
3. Make sure `npm run env start` works with different ports and that the containers work as you'd expect them to (correct ports in URLs, etc).
